packages/gatsby-telemetry/src/event-storage.js

```diff
@@ -0,0 +1,64 @@
+const Store = require(`./store`)
+const fetch = require(`node-fetch`)
+
+/* The events data collection is a spooled process that
+ * buffers events to a local fs based buffer
+ * which then is asynchronously flushed to the server.
+ * This both increases the fault tolerancy and allows collection
+ * to continue even when working offline.
+ */
+module.exports = class EventStorage {
+  store = new Store()
+  debugEvents = !!process.env.GATSBY_TELEMETRY_DEBUG
+  disabled = !!process.env.GATSBY_TELEMETRY_OPTOUT
+
+  addEvent(event) {
+    if (this.disabled) {
+      return
+    }
+
+    if (this.debugEvents) {
```

Comment:
wondering if debugEvents should log and capture it instead of just logging.

Reply:
This felt more natural when implementing it, it has the benefit of having ability of introspect the events w/o sending any data. And it's the same behaviour as with homebrew analytics.